### PR TITLE
dbr quant: support for custom leaf modules, part 2/x

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70372 dbr quant: add limited support for `torch.nn.ModuleList`
* #70349 dbr quant support for custom leaf modules, part 3/x
* **#70335 dbr quant: support for custom leaf modules, part 2/x**
* #70330 dbr quant: support for custom leaf modules, part 1/x
* #70329 dbr quant: unbreak case when child module not returning any outputs
* #70328 dbr quant: support functional linear without bias
* #70257 dbr quant: support dynamic linear
* #70256 dbr quant: do not attach auto_quant_state to observers

Summary:

Adds test case that functions are not quantized inside custom leaf modules.
No logic change needed as it already works correctly.

Note: FX scripting rewriter does not go into modules without auto-quant,
which is why we are using torch.jit.trace to look at the graph.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR.test_prepare_custom_config_dict_non_traceable_module_class
```

Differential Revision: [D33286370](https://our.internmc.facebook.com/intern/diff/D33286370)

### CODE REVIEW

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33286370).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33286370).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33286370).

File: test/quantization/dbr/test_quantize_dbr.py [hunk 1]
```diff
@@ -624,15 +625,17 @@ def forward(self, x):
     def test_prepare_custom_config_dict_non_traceable_module_class(self):
```
Comment (jerryzh168) [inline test/quantization/dbr/test_quantize_dbr.py:625]: nit: wondering if we want to break the test into smaller ones so that it's easier to read

Comment (jerryzh168) [reply test/quantization/dbr/test_quantize_dbr.py:625]: and the first test can be simpler

**Review by jerryzh168 — APPROVED** [review]

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33286370).

**Pull request closed without merging**